modules/stores/tree.js

```diff
@@ -565,6 +569,85 @@ const setValue = (state, path, delta, value, valueType, config, asyncListValues,
   return {tree: state, isInternalValueChange};
 };
 
+// this is a copy of setValue function witch changes required for LHS
+const setLhsValue = (state, path, delta, value, valueType, config, asyncListValues, __isInternal) => {
+  const {fieldSeparator, showErrorMessage} = config.settings;
+  let isInternalValueChange;
+  const valueSrc = state.getIn(expandTreePath(path, "properties", "lhsValueSrc", delta + "")) || null;
+  if (valueSrc === "field" && Array.isArray(value))
+    value = value.join(fieldSeparator);
+
+  const field = state.getIn(expandTreePath(path, "properties", "field")) || null;
+  const operator = state.getIn(expandTreePath(path, "properties", "operator")) || null;
+  const operatorConfig = getOperatorConfig(config, operator, field);
+  const operatorCardinality = operator ? defaultValue(operatorConfig.cardinality, 1) : null;
+
+  const isEndValue = false;
+  const canFix = false;
+  const calculatedValueType = valueType || calculateValueType(value, valueSrc, config);
+  const [validateError, fixedValue] = validateValue(
```

Comment:
Shouldn't that use lhs too?

Reply:
It does - in LHS `Widget` component is paramterized with `lhsValue` and `setLhsValue`. The function is then called with the `value` parameter which in LHS is `lhsValue`